src/Shared/Network/Connection.cs

```diff
@@ -59,6 +71,14 @@ public Connection()
 
 			this.State = ConnectionState.Open;
 			this.Address = "?:?";
+			this.IntegritySeed = _random.Next();
```

Comment:
You could use `RandomProvider.Get().Next()` here, saves you from creating an instance yourself, especially since it's never used again.